sdk/scala-sdk-testkit/src/main/scala/kalix/scalasdk/testkit/impl/TestKitMockRegistryImpl.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.scalasdk.testkit.impl
+
+import kalix.scalasdk.testkit.TestKitMockRegistry
+
+private[kalix] class TestKitMockRegistryImpl(var mocks: Set[Any] = Set.empty) extends TestKitMockRegistry {
+
+  override def withMock[T](instance: T): TestKitMockRegistry = {
+    mocks = mocks + instance
+    this
+  }
+
+  override def get[T](clazz: Class[T]): Option[T] = mocks.collectFirst {
+    case m if m.getClass.getAnnotatedInterfaces.exists(_.getType == clazz) => clazz.cast(m)
+  }
```

Comment:
Notice how on the `scalasdk` I'm using a `Set` and not a `Map` (used in the `javasdk)`... The end goal would be to choose between one or the other version and make it consistent on both sdks ... I've kept both options for the sake of the review because I'm not quite sure. Although this version with the `Set` allows the user to do only `TestKitMockRegistry.withMock(mockA)` instead of `TestKitMockRegistry.withMock(A.class, mockA)` I wonder if this convoluted logic is a bit too risky. @johanandren @octonato